SwiftUIHomeworks/SwiftUIHomeworks/Screens/CartView.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  CartView.swift
+//  SwiftUIHomeworks
+//
+//  Created by Aleksandr Dorofeev on 21.01.2023.
+//
+
+import SwiftUI
+
+/// Cart view
+struct CartView: View {
+  // MARK: - Private Constants
+  
+  private enum Constants {
+    static let emptyCartTitle = "Cart is empty"
+    static let navigationTitle = "Cart"
+    static let payButtonTitle = "Pay"
+    static let infoAboutOrderTitle = "Info about order"
+    static let confirmChoiceMessage = "Confirm your choice\n"
+    static let opacityZeroValue: CGFloat = 0
+    static let opacityOneValue: CGFloat = 1
+    static let weightButtonValue: CGFloat = 100
+    static let heightButtonValue: CGFloat = 50
+    static let rubTitle = "RUB"
+    static let daysTitle = "days"
+  }
+  
+  // MARK: - Body
+  
+  var body: some View {
+    NavigationView {
+      ZStack {
+        Text(Constants.emptyCartTitle)
+        buySectionView
+          .opacity(purchaseViewModel.isBuying ? Constants.opacityOneValue : Constants.opacityZeroValue)
+      }
+      .navigationTitle(Constants.navigationTitle)
+    }
+  }
+  
+  // MARK: - Private properties
+  
+  @EnvironmentObject private var purchaseViewModel: PurchaseViewModel
+  @State private var isConfirmationAlertShown = false {
+    didSet {
+      purchaseViewModel.isEnabled = true
+    }
+  }
+  
+  private var buySectionView: some View {
+    Form {
+      Section {
+        HStack {
+          Text("\(purchaseViewModel.limitationValue) \(Constants.daysTitle)")
+          Spacer()
+          Text("\(purchaseViewModel.priceValue) \(Constants.rubTitle)")
+        }
+        Spacer()
+        Button {
+          self.isConfirmationAlertShown.toggle()
+        } label: {
+          Text(Constants.payButtonTitle)
+        }.alert(isPresented: $isConfirmationAlertShown, content: {
+          Alert(title: Text(Constants.infoAboutOrderTitle),
+                message: Text("\(Constants.confirmChoiceMessage) \(purchaseViewModel.limitationValue) \(Constants.daysTitle) : \(purchaseViewModel.priceValue) \(Constants.rubTitle)"),
+                primaryButton: .destructive(Text(Constants.payButtonTitle),
+                                            action: {
+            self.purchaseViewModel.isBuying.toggle()
+          }),
+                secondaryButton: .cancel())
+        })
+        .frame(width: Constants.weightButtonValue, height: Constants.heightButtonValue)
+        .background(Color.blue)
+        .foregroundColor(.white)
```

Comment:
```suggestion
        Button {
          self.isConfirmationAlertShown.toggle()
        } label: {
          Text(Constants.payButtonTitle)
        }.alert(isPresented: $isConfirmationAlertShown, content: {
          Alert(title: Text(Constants.infoAboutOrderTitle),
                message: Text("\(Constants.confirmChoiceMessage) \(purchaseViewModel.limitationValue) \(Constants.daysTitle) : \(purchaseViewModel.priceValue) \(Constants.rubTitle)"),
                primaryButton: .destructive(Text(Constants.payButtonTitle),
                                            action: {
            self.purchaseViewModel.isBuying.toggle()
          }),
                secondaryButton: .cancel())
        })
        .frame(width: Constants.weightButtonValue, height: Constants.heightButtonValue)
        .background(Color.blue)
        .foregroundColor(.white)
```
и это тоже, как по мне много всего в одном свойстве